File: chroma-league/src/test/java/net/booone/chromaleague/hud/parts/resource/RedFuryBarTest.java
package net.booone.chromaleague.hud.parts.resource;

import net.booone.chromaleague.IntSteps;
import net.booone.chromaleague.hud.AnimationTester;
import org.junit.jupiter.api.Test;

class RedFuryBarTest extends AbstractResourceTest {
    @Test
    void testRedFuryBar() {
        final IntSteps intSteps = new IntSteps(0, 100, 5);
        final RedFuryBar yoneCloneBar = new RedFuryBar();

        new AnimationTester()
                .withBeforeIterationAction(i -> mockResource(intSteps.nextInt(), 100))
                .testAnimation(yoneCloneBar, 40);
    }
}

File: test/hud/parts/resource/RedFuryBar.test.ts
import {describe, test} from 'vitest';
import {IntSteps} from '../../../IntSteps.js';
import {AnimationTester} from '../../../AnimationTester.js';
import {RedFuryBar} from '../../../../src/hud/parts/resource/RedFuryBar.js';
import {mockResource, setupResourceTest} from './AbstractResourceTest.js';

/** Integration test for red fury resource bar on actual hardware. */
describe('RedFury Integration', () => {
    setupResourceTest();

    test('testRedFuryBar', async () => {
        const intSteps = new IntSteps(0, 100, 5);
        const redFuryBar = new RedFuryBar();

        await new AnimationTester()
            .withBeforeIterationAction(() => mockResource(intSteps.nextInt(), 100))
            .testAnimation(redFuryBar, 40);
    }, 30000);
});
